### PR TITLE
ByteSize conversion

### PR DESCRIPTION
**Pull request by dgasmith**
## Change Summary

Allows the conversation of a ByteSize object to another unit.

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [x] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)

### CODE REVIEW

**Comment by samuelcolvin** [comment]
Lgtm.

Haven't looked but I bet the errors are related to single quotes.

**Comment by dgasmith** [comment]
`make lint` passes for me locally so I seem to forget about this issue. Any ideas why this would be?

**Comment by samuelcolvin** [comment]
`make lint` definitely shouldn't fail if you have double quotes in `pydantic/` or `tests/`

**Comment by dgasmith** [comment]
Odd, it fails on Travis? This seems like the correct behavior.

```
8.12s$ make lint
572flake8 pydantic/ tests/
573tests/test_types.py:1874:20: Q000 Remove bad quotes
574tests/test_types.py:1876:36: Q000 Remove bad quotes
```

**Comment by samuelcolvin** [comment]
> `make lint` definitely shouldn't fail if you have double quotes in `pydantic/` or `tests/`

My mistake, I meant "**should** fail if you have double quotes" :face_with_head_bandage: :gun: :pray:

File: tests/test_types.py [hunk 1]
```diff
@@ -1866,6 +1867,17 @@ class Model(BaseModel):
     assert m.size.human_readable(decimal=True) == human_dec
 
 
+def test_bytesize_to():
+    class Model(BaseModel):
+        size: ByteSize
+
+    m = Model(size='1GiB')
+
+    assert pytest.approx(m.size.to('MiB')) == 1024
```
Comment (samuelcolvin) [inline tests/test_types.py]:
```suggestion
    assert m.size.to('MiB') == pytest.approx(1024)
```

reads better and is what's suggested by [the docs](https://docs.pytest.org/en/latest/reference.html#pytest-approx)

File: tests/test_types.py [hunk 2]
```diff
@@ -1866,6 +1867,17 @@ class Model(BaseModel):
...
+    assert pytest.approx(m.size.to('MB')) == 1073.741824
```
Comment (samuelcolvin) [inline tests/test_types.py]:
```suggestion
    assert m.size.to('MB') == pytest.approx(1073.741824)
```

File: tests/test_types.py [hunk 3]
```diff
@@ -1866,6 +1867,17 @@ class Model(BaseModel):
...
+    assert pytest.approx(m.size.to('TiB')) == 0.0009765625
```
Comment (samuelcolvin) [inline tests/test_types.py]:
```suggestion
    assert m.size.to('TiB') == pytest.approx(0.0009765625)
```

**Comment by samuelcolvin** [comment]
otherwise LGTM.

**Comment by dgasmith** [comment]
Hah :) Well, I don't know then. I'll try to keep a sharper lookout myself in the future.

```bash
>>> cat pydantic/version.py | tail -n1
VERSION = StrictVersion("1.1")
>>> flake8 --config=setup.cfg pydantic/version.py
```

**Comment by samuelcolvin** [comment]
I'm not sure what you're saying about `version.py`, it definitely uses single quotes:

https://github.com/samuelcolvin/pydantic/blob/06b2204a6be2a31c57c4ebe950c70a3771bed062/pydantic/version.py#L1-L5

**Comment by dgasmith** [comment]
I edited the file locally as a demo.

**Comment by samuelcolvin** [comment]
okay, hope it's clear now.

Thanks for this.

**Pull request merged**